### ISSUE TITLE
How can I tell which is the primary monitor in the monitors array?

### ISSUE THREAD

**Issue by mgabor3141** [BoboTiG/python-mss#153]

**Comment by BoboTiG** [comment]
There is no proper way. My guess would be that the first monitor retrieved is the primary one, but I may be wrong.

**Comment by halldorfannar** [comment]
Is this for Windows?  Because I would like to propose this enhancement:

# Expose primary monitor flag on Windows
## Summary

Currently, there's no clean way to identify which monitor is the Windows primary monitor. Users must infer it by checking for coordinates `(left=0, top=0)`, which works but isn't explicit.

## Proposal

Expose the `MONITORINFOF_PRIMARY` flag from the Windows API by:

1. Calling `GetMonitorInfo` in the `EnumDisplayMonitors` callback
2. Checking `mi.dwFlags & MONITORINFOF_PRIMARY`
3. Either:
   - Add an `"is_primary": bool` field to each monitor dict, or
   - Provide a `primary_monitor` property on the MSS class

## Current Workaround

```python
def get_primary_monitor(sct):
    """Find the primary monitor (the one at origin 0,0)."""
    for monitor in sct.monitors[1:]:  # Skip monitors[0] which is "all"
        if monitor["left"] == 0 and monitor["top"] == 0:
            return monitor
    return sct.monitors[1]  # Fallback
```

## Platform

Windows only (other platforms may have similar concepts but different APIs).

**Comment by BoboTiG** [comment]
Lets start with Windows then. Adding a `"is_primary": bool` field to each monitor dict seems good to me.

**Comment by BoboTiG** [comment]
And, we could provide the property as well, it will lower the work for all downstream projects.

**Comment by halldorfannar** [comment]
Yes, doing both sounds good. Please assign this issue to me.

**Comment by halldorfannar** [comment]
I will probably do something like this for the property:

```python
@property
def primary_monitor(self) -> Monitor:
    """Return the primary monitor."""
    for monitor in self._monitors[1:]:
        if monitor.get("is_primary"):
            return monitor
    return self._monitors[1]  # Fallback to first
```

That way it will work even for Linux without "is_primary" field (we will eventually add it though).  I could also have it return an exception with NotImplemented, if preferred.

**Comment by BoboTiG** [comment]
I would vote for a default value on other OS rather than raising an exception. You POC seems good!

**Comment by halldorfannar** [comment]
@jholveck and I discussed this task this morning and one thing that we would like your verdict on is the following:

So far the monitor object has just been a dict.  This makes a lot of sense because it has contained homogenous data (screen coordinates).  With the addition of `is_primary` boolean that changes.  We also discussed that eventually it would be nice to include device names in the monitor data as well (makes identification easier).  At that point we would have screen coordinates, boolean and string in the dict.  This naturally smells like a class object at that point.  We can make `__get_item__` accessor so that dict like access would still work.  That way the change would be backwards compatible.  Or do you prefer that we keep it as a dict?

What do you think?

**Comment by BoboTiG** [comment]
I like the idea of a class object with `__get_item__`!

**Comment by jholveck** [comment]
@halldorfannar , once your PR gets merged, can you bounce this to me to do the Linux side based on your code?

**Comment by halldorfannar** [comment]
Yes, @jholveck.  Once merged I will pass this hot potato, I mean issue, over to you.
